server/midddleware/metrics.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2022 Tigris Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package middleware
+
+import (
+	"context"
+	"github.com/tigrisdata/tigris/server/metrics"
+	"github.com/uber-go/tally"
+	"google.golang.org/grpc"
+	"strings"
+)
+
+type grpcCallData struct {
+	grpcMethod string
+	grpcType   string
+}
+
+func newGrpcCallData(grpcMethod string, grpcType string) grpcCallData {
+	return grpcCallData{grpcMethod: grpcMethod, grpcType: grpcType}
+}
+
+func (g *grpcCallData) getTags() map[string]string {
```

Comment:
So as tags are expensive to create with each request should we think about defining some of the known grpc method tags already at the top? Mainly for `Insert/Replace/Update/Delete/Read`. If tag is already defined above if use it otherwise we create it in this method. This reduce tag creation for common APIs. @efirs @pboros ?

Reply:
We had a discussion about this, in the next diff, I will work towards making the counters and the tags both static (create them at the time of creating the services).